### DISCUSSION TITLE
Send individual information to each participant

### DISCUSSION THREAD

**Discussion by svollrath** [rubenarslan/formr.org#548] [Q&A]
I am conducting a study on feedback acceptance. Participants are asked to write a short text and an AI will generate individual feedback for each one. A week later, each participant should receive their personalized feedback and complete a questionnaire.

I can manually extract the written texts from formR and provide them to the AI. However, I need a method to ensure each participant receives their individual feedback. Ideally, participants should receive a link via email that directs them to their feedback displayed along with the questionnaire. 

Does anyone have experience with presenting individual information/feedback to participants or know of any tools that can automate this process effectively?

For clarification: the feedback is generated by an AI based on the participants' texts, not feedback generated from the formR data for analysis with R.

**Comment by dfeistauer** [comment]
I tried it a bit in R. This is the code in R. 
You need to try it in formr to see if it is useful or if it needs changes.

the csv looks like this
<img width="484" alt="image" src="https://github.com/rubenarslan/formr.org/assets/44641119/d9bd88ce-37df-48fb-93a3-316b4d1ce8c5">

**Stuff for a new survey:**
stuff for a calculate-type in column value
```
# preparation of the download
url <- "https://uni-muenster.sciebo.de/s/faDe4e5BTeoqh4d/download"
temp <- tempfile(fileext = ".csv")
download.file(url, temp, mode = "wb")

# load csv-file 
daten <- read.csv2(temp)

# remove unnessecary stuff
unlink(temp)
rm(temp, url)

daten
```

stuff for a second calculate-type in column value
```
# read Session-ID from get-Parameter - this is send in the reminder to the subjects too
# todo in Formr
session_id <- "556zhgfa" # expample
```

preparation of showing the feedback
Calculate-Type (no. 3 & 4)
```
Antwort <- daten[daten$sessionCode == session_id, "VPAntwort"]
Feedback <- daten[daten$sessionCode == session_id, "Feedback"]
```

use these information in the note-types
in column Label of the survey

**End of this survey**

You need to incoperate maybe the survey names in the r-code.

**Comment by dfeistauer** [comment]
Possible Run Structure for formr

Module 10: participant answers a question.
Module 20: Email to the experiment management, informing them that someone has participated.
Module 30: Break for 6 days.
    • You need to copy the information into the CSV file from above.
Module 40: Break for the remaining time.
    • Read the time from the CSV file from Module 30. This might be a bit tricky because you need to import the csv-File from before. Maybe you need for this another survey with a calculate-Type. This would be Module 35.
Module 50: Email to the participant, informing them that the experiment continues.
    • Consider how quickly the participant should respond and what to do if they do not respond.
Module 60: Survey with code from above.
any stuff between
Module 1000: End.
